src/components/panel/panel.js

```diff
@@ -721,6 +758,9 @@ function MdPanelService($rootElement, $rootScope, $injector, $window) {
   /** @private {!Object<string, !MdPanelRef>} */
   this._trackedPanels = {};
 
+  /** @private {!Object<string, !Object<{panels:Array<!MdPanelRef>, maxOpen:number}>>} */
```

Comment:
{!Object<string, {panels: !Array<!MdPanelRef>, maxOpen: number}>}

This is the correct type. You are explict about the object keys, so you don't write Object. Also fixed nullability on Array.